wordpress-mobile/WordPress-iOS
# Drop iOS 14 Support: Remove deprecated UIDocumentPickerViewController API

Part of https://github.com/wordpress-mobile/WordPress-iOS/issues/20860

Remove deprecated `UIDocumentPickerViewController` API usage

To test:

- Open Gutenberg or Aztec (best to test both)
- Add File
- Select one of the available files
- Verify that the files got uploaded

The list of available file formats can differ based on a blog. In case of my test blog, it was the following:

```
  - 0 : "dyn.ah62d4rv4ge80c65d"
  - 1 : "com.microsoft.excel.xls"
  - 2 : "public.jpeg"
  - 3 : "com.microsoft.word.doc"
  - 4 : "org.oasis-open.opendocument.text"
  - 5 : "org.openxmlformats.presentationml.slideshow"
  - 6 : "org.openxmlformats.wordprocessingml.document"
  - 7 : "com.adobe.pdf"
  - 8 : "public.png"
  - 9 : "public.heif"
  - 10 : "com.apple.iwork.keynote.sffkey"
  - 11 : "com.microsoft.powerpoint.ppt"
  - 12 : "org.openxmlformats.presentationml.presentation"
  - 13 : "com.compuserve.gif"
  - 14 : "org.openxmlformats.spreadsheetml.sheet"
  - 15 : "com.microsoft.powerpoint.pps"
  - 16 : "public.heic"
  - 17 : "org.webmproject.webp"
  - 18 : "public.jpeg"
```

The before/after should be functionality equivalent as the only thing that changes is that the app uses the `UTType` API instead of a `String`-based API. There should be no need to test every single one of theses formats.

## Regression Notes
1. Potential unintended areas of impact


2. What I did to test those areas of impact (or what existing automated tests I relied on)


3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:


UI Changes testing checklist:
